gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -53,10 +57,14 @@
   private static final Logger LOG = LoggerFactory.getLogger(BaseDataPublisher.class);
 
   protected final List<FileSystem> fss = Lists.newArrayList();
+  protected final Closer closer;
   protected int numBranches;
+  protected int parallelRunnerThreads;
+  protected Map<String, ParallelRunner> parallelRunners;
```

Comment:
I suggest doing `protected final Map<String, ParallelRunner> parallelRunners = Maps.newHashMap();`. By doing this, `parallelRunners` is guaranteed to be non-null when it's used.